src/factories/locationFactory.ts

```diff
@@ -36,6 +39,32 @@ export namespace LocationFactory {
             uri = Utilities.combineUriWithUriBase(physLocation.artifactLocation.uri, uriBase);
 
             if (uri) {
+                // Check for artifact content and create our artifact content URI.
+                // In order to have artifact content, the log must have a valid artifact index
+                // (represented by artifact location index) and the "contents" property of the artifact
+                // must be defined and have one of the "binary, text, or rendered" properties set.
+                if (physLocation.artifactLocation.index !== undefined) {
+                    fileName = uri.toString(true).substring(uri.toString(true).lastIndexOf('/') + 1);
+
+                    const artifactContentUri: Uri | undefined = ArtifactContentFileSystemProvider.tryCreateUri(sarifLog, Uri.file(runInfo.sarifFileFullPath), uri, runInfo.runIndex, physLocation.artifactLocation.index);
+                    if (artifactContentUri) {
+                        uri = artifactContentUri;
+                        mappedToLocalPath = true;
+
+                        // For binary content, we can map the byte offset and length to the "markdown" it will be converted to
+                        // for user display.
+                        const binaryContentRenderer: BinaryContentRenderer | undefined = BinaryContentRenderer.tryCreateFromLog(sarifLog, runInfo.runIndex, physLocation.artifactLocation.index);
+                        if (binaryContentRenderer &&
```

Comment:
Internally we need to keep the translated region. However do we show users the original row/col numbers (in the context of the binary), or the translated ones (in the context of markdown)?

Reply:
We use VSCode's related diagnostic information for that... 

We have many options here.

---
In reply to: [411630299](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411630299) [](ancestors = 411630299)

Reply:
We can open an issue later if we deem it to be necessary.

---
In reply to: [411702385](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411702385) [](ancestors = 411702385,411630299)